x-pack/plugins/monitoring/public/alerts/lib/security_toasts.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSpacer, EuiLink, EuiCode, EuiText } from '@elastic/eui';
+import { Legacy } from '../../legacy_shims';
+import { toMountPoint } from '../../../../../../src/plugins/kibana_react/public';
+
+export interface AlertingFrameworkHealth {
+  isSufficientlySecure: boolean;
+  hasPermanentEncryptionKey: boolean;
+}
+
+const showTlsAndEncryptionError = () => {
+  const { ELASTIC_WEBSITE_URL, DOC_LINK_VERSION } = Legacy.shims.docLinks;
+
+  Legacy.shims.toastNotifications.addWarning({
+    title: toMountPoint(
+      <FormattedMessage
+        id="xpack.monitoring.healthCheck.tlsAndEncryptionErrorTitle"
+        defaultMessage="Additional setup required"
+      />
+    ),
+    text: toMountPoint(
+      <div>
+        <p>
+          {i18n.translate('xpack.monitoring.healthCheck.tlsAndEncryptionError', {
+            defaultMessage: `You must enable Transport Layer Security between Kibana and Elasticsearch 
+              and configure an encryption key in your kibana.yml file to use the Alerting feature.`,
+          })}
+        </p>
+        <EuiSpacer />
+        <EuiLink
+          href={`${ELASTIC_WEBSITE_URL}guide/en/kibana/${DOC_LINK_VERSION}/alert-action-settings-kb.html#general-alert-action-settings`}
+          external
+          target="_blank"
+        >
+          {i18n.translate('xpack.monitoring.healthCheck.encryptionErrorAction', {
+            defaultMessage: 'Learn how.',
+          })}
+        </EuiLink>
+      </div>
+    ),
+  });
+};
+
+const showEncryptionError = () => {
+  const { ELASTIC_WEBSITE_URL, DOC_LINK_VERSION } = Legacy.shims.docLinks;
+
+  Legacy.shims.toastNotifications.addWarning(
+    {
+      title: toMountPoint(
+        <FormattedMessage
+          id="xpack.monitoring.healthCheck.encryptionErrorTitle"
+          defaultMessage="You must set an encryption key"
+        />
+      ),
+      text: toMountPoint(
+        <div role="banner">
+          {i18n.translate('xpack.monitoring.healthCheck.encryptionErrorBeforeKey', {
+            defaultMessage: 'To create an alert, set a value for ',
```

Comment:
Again, like before, the user will not understand what this means because they aren't consciously enabling or interacting with alerts

Reply:
Ah you're right!

Wondering if we should just use one generic message from https://github.com/elastic/kibana/pull/71846/files#r455776054 if any of the two requirements (`isSufficientlySecure`, and `hasPermanentEncryptionKey ) are false

Reply:
Yea I'm not sure what we want to say here. Something like "out of the box alerts require x, y, z. See {link} on how to configure this". Honestly, I'd default to @ravikesarwani @lcawl @gchaps for help here on the messaging

Reply:
Worked with @lcawl on the text.  Does this work:

Alerting requires Transport Layer Security between Kibana and Elasticsearch and an encryption key in your kibana.yml file.  Learn how to enable TLS.

where "Lean how to enable TLS" links to the documentation.

Reply:
Yea, I think that works. i'll open a new PR for this

Reply:
See https://github.com/elastic/kibana/pull/72310